pkg/volume/hostpath/host_path.go

```diff
@@ -470,29 +470,46 @@ func checkTypeInternal(ftc hostPathTypeChecker, pathType *v1.HostPathType) error
 }
 
 // makeDir creates a new directory.
-// If pathname already exists as a directory, no error is returned.
+// If pathname already exists as a directory and with proper permissions, no error is returned.
 // If pathname already exists as a file, an error is returned.
 func makeDir(pathname string) error {
-	err := os.MkdirAll(pathname, os.FileMode(0755))
+	// TODO make "mode" a parameter for the function
+	const mode = os.FileMode(0755)
+	err := os.MkdirAll(pathname, mode)
 	if err != nil {
 		if !os.IsExist(err) {
 			return err
 		}
+		// check if directory that exists have the required persmissions
+		if dir, _ := os.Stat(pathname); dir.Mode() != mode {
+			return fmt.Errorf("%s exists and has wrong persmission, wanted %d", pathname, mode)
+		}
 	}
 	return nil
 }
 
 // makeFile creates an empty file.
-// If pathname already exists, whether a file or directory, no error is returned.
+// If pathname already exists, and is a file no error is returned.
+// If pathname already exists, and is a directory, error is returned.
 func makeFile(pathname string) error {
-	f, err := os.OpenFile(pathname, os.O_CREATE, os.FileMode(0644))
+	// TODO make "mode" a parameter for the function
+	const mode = os.FileMode(0644)
+	f, err := os.OpenFile(pathname, os.O_CREATE, mode)
```

Comment:
Hi, @vaibhavk.
what about if your code will looks like this one?
```go
func makeFile(pathname string) error {
  fStat, err := os.Stat(pathname)
  if os.IsNotExists(err) {
    return err
  }
  if fStat.Mode().IsDir() {
    return fmt.Errorf("%s is a directory", pathname)
  }
  const mode = os.FileMode(0644)
  f, err := os.OpenFile(pathname, os.O_CREATE, mode)
  //check that file exists and have required permissions
  if (os.IsPermission(err)) {
    return fmt.Errorf("%s exists and has wrong persmission, wanted %d", pathname, mode)
  }
  f.Close()
  return nil
}
```
I'm not sure that my code work, because I've never wrote a piece of code in GO, however looks better, isn't it?

Reply:
Surely, it's more readable this way, I'll make changes accordingly

Reply:
While at it, notice the typo in `persmission` :)

That makes a really funny word in Finnish, I like it.